flatpak-proxy.c

```diff
@@ -2556,63 +2611,44 @@ auth_line_is_valid (guint8 *line, guint8 *line_end)
   return TRUE;
 }
 
-static gboolean
-auth_line_is_begin (guint8 *line)
-{
-  guint8 next_char;
-
-  if (!g_str_has_prefix ((char *) line, AUTH_BEGIN))
-    return FALSE;
-
-  /* dbus-daemon accepts either nothing, or a whitespace followed by anything as end of auth */
-  next_char = line[strlen (AUTH_BEGIN)];
-  return next_char == 0 ||
-         next_char == ' ' ||
-         next_char == '\t';
-}
-
 static gssize
 find_auth_end (FlatpakProxyClient *client, Buffer *buffer)
 {
-  goffset offset = 0;
-  gsize original_size = client->auth_buffer->len;
-
-  /* Add the new data to the remaining data from last iteration */
-  g_byte_array_append (client->auth_buffer, buffer->data, buffer->pos);
-
-  while (TRUE)
-    {
-      guint8 *line_start = client->auth_buffer->data + offset;
-      gsize remaining_data = client->auth_buffer->len - offset;
-      guint8 *line_end;
-
-      line_end = memmem (line_start, remaining_data,
-                         AUTH_LINE_SENTINEL, strlen (AUTH_LINE_SENTINEL));
-      if (line_end) /* Found end of line */
-        {
-          offset = (line_end + strlen (AUTH_LINE_SENTINEL) - client->auth_buffer->data);
-
-          if (!auth_line_is_valid (line_start, line_end))
-            return FIND_AUTH_END_ABORT;
+  gssize new_state = client->auth_state;
```

Comment:
This code relies on the buffer being a complete line (and nothing more than a line). The later is now true, but if you handle errors per the above that is no longer true. So you would need to handle cases where the CRLF is not yet read.